### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by BirSikanderMahajan**

### CODE REVIEW

**Review by RomanSyrychLeapwork — COMMENTED** [review]
@BirSikanderMahajan  Please, have a look at those comments
@AntonLiakhovich Could you also have a look on my question in one of the comments?

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 1]
```diff
@@ -0,0 +1,804 @@
...
+                case 200:
+                    JsonParser parser = new JsonParser();
+                    JsonArray jsonScheduleList = parser.parse(response.getResponseBody()).getAsJsonArray();
+
+                    for (String rawSchedule : rawScheduleList) {
+                        boolean successfullyMapped = false;
+                        for (JsonElement jsonScheduleElement : jsonScheduleList) {
+                            JsonObject jsonSchedule = jsonScheduleElement.getAsJsonObject();
+
+                            if(jsonSchedule.get("Type").getAsString().contentEquals("TemporaryScheduleInfo")) continue;
```
[the remark this answers is not in the document]
Comment (AntonLiakhovich) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:186]: yes, we don't have it. this logic can be depricated

File: target/generated-sources/localizer/io/jenkins/plugins/sample/Messages.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ * Generated localization support class.
+ * 
+ */
+@SuppressWarnings({
+    "",
+    "PMD",
+    "all"
+})
+@Restricted(org.kohsuke.accmod.restrictions.NoExternalUse.class)
+public class Messages {
```
[the remark this answers is not in the document]
Comment (BirSikanderMahajan) [reply target/generated-sources/localizer/io/jenkins/plugins/sample/Messages.java:20]: yes the Messages class is being used in PluginHandler class.

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:186]: Deleted this piece of code

File: README.md [hunk 3]
```diff
@@ -1,2 +1,79 @@
-# jenkins
-Integration with Jenkins
+# Leapwork Integration
+This is Leapwork plugin for Jenkins (version 2.60.3 or later)
```
[the remark this answers is not in the document]
Comment (BirSikanderMahajan) [reply README.md:2]: This is the recommended version from jenkins

https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/

File: target/generated-sources/localizer/io/jenkins/plugins/sample/Messages.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (RomanSyrychLeapwork) [reply target/generated-sources/localizer/io/jenkins/plugins/sample/Messages.java:20]: Are you sure about it? It look like this Messages class is using in PluginHandler class. Will the solution be built if you remove current class?
image

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (RomanSyrychLeapwork) [reply README.md:2]: @BirSikanderMahajan That's ok, but we need to update the description in ReadMe here like "(version 2.277.1 or later)"

File: target/generated-sources/localizer/io/jenkins/plugins/sample/Messages.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (BirSikanderMahajan) [reply target/generated-sources/localizer/io/jenkins/plugins/sample/Messages.java:20]: Yes I'm Sure about it, Plugin Handler class is using Messages class.
In Messages class we have all the Strings (API, Messages, etc) So building plugin without this class wouldn't work.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (BirSikanderMahajan) [reply README.md:2]: updated in the same branch

**Pull request closed without merging**